src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.cs

```diff
@@ -341,7 +352,8 @@ private void MakeFrames(ArrayBuilder<ClosureDebugInfo> closureDebugInfo)
 
                 foreach (var captured in capturedVars)
                 {
-                    if (!_analysis.VariableScope.TryGetValue(captured, out BoundNode captureScope))
+                    var declarationScope = Analysis.GetVariableDeclarationScope(scope, captured);
+                    if (declarationScope == null)
```

Comment:
What does this mean when it's null? The variable isn't found? Is that valid? (Concern is if we should be silently eating this (it's a legitimate case) or, `Debug.Assert(false)`/`throw UnreachableException`)

(Alternatively, I see that it's likely just copying behavior of the old implementation - we may want to revisit this once you move onto the phase of improvement instead of pure behavior-preserving refactoring)